### PR TITLE
Add tests for the remaining part of the codebase

### PR DESCRIPTION
**Pull request by sdglitched**
Add tests for the remaining part of the codebase

- Add `pragma: no cover` for the main function which initialize the user interface
- Remove exception handling from the graphics resource modifier as not required
- Add `pragma: no cover` for constructor initialisation of Facility and Dialog to exclude it from the `coverage`
- Remove parameter of QMessageBox from MainWindow as it already getting initialised by the constructor of the base class in the Dialog
- Remove hpak.py as not getting used in the code functionality

### CODE REVIEW

File: gi_loadouts/face/wind/main.py [hunk 1]
```diff
@@ -9,12 +9,19 @@
...
         self.setWindowTitle(f"Loadouts for Genshin Impact v{__versdata__}")
         self.initialize_events()
         self.initialize_elements()
-        make_temp_file()
+        try:
+            make_temp_file()
+        except Exception as expt:
+            self.show_dialog(
+                QMessageBox.Information,
+                "Initialisation Failed",
```
Comment (gridhead) [inline gi_loadouts/face/wind/main.py:22]:
```suggestion
                "Initialization failed",
```

File: test/face/scan/test_scan.py [hunk 2]
```diff
@@ -481,10 +496,93 @@ def test_scan_import_arti(scantest, qtbot, mocker, _) -> None:
...
+)
+def test_scan_import_arti_flty(scantest, qtbot, _) -> None:
+    """
+    Attempt importing faulty artifact information in MainWindow
+
+    :return:
+    """
+
+    """
+    Set values in scan inerface before before transfering it to MainWindow
```
Comment (gridhead) [inline test/face/scan/test_scan.py:522]:
```suggestion
    Set values in scan interface before before transferring it to MainWindow
```

File: test/face/scan/test_scan.py
Comment (gridhead) [inline test/face/scan/test_scan.py:1]: In line #433 (Ref. https://github.com/gridhead/gi-loadouts/pull/184/files#diff-40a1e8e1617fc1a62c91a37bc092fc5b97ebd06689b9ef0bec180860d9a979acR433),

```python
        pytest.param(None, id="face.scan.rule: Clearing snapshot")
```

In line #472 (Ref. https://github.com/gridhead/gi-loadouts/pull/184/files#diff-40a1e8e1617fc1a62c91a37bc092fc5b97ebd06689b9ef0bec180860d9a979acR472),

```python
    Create the Tesseract training data
```

**Review by gridhead — CHANGES_REQUESTED** [review]
Let us get started with the following changes first.

**Comment by sdglitched** [comment]
Applied all the changes as per suggestion! ;)

**Review by gridhead — APPROVED** [review]
The GitHub Actions CI needs to be looked into. 

But apart from that, this looks good to me.

Thanks for getting the coverage to a grand 100%! 🎉

**Pull request merged**